### PR TITLE
Feature/bycrypt

### PR DESCRIPTION
**Pull request by Alexandria**
## Reference

[37](https://crema.atlassian.net/browse/BA-37)

## Overview

I created a sign up path in my api that encrypts the users password before it is put into the database.
I modified the login path to check to see if entered password matches the encrypted one  in the data base.   

### Acceptance Criteria

- [x] Passwords are no longer stored in plain text in the database

#### Screenshot of the users in the data base.
Screen Shot 2019-07-01 at 11 57 00 AM

## Reviewers' Checklist (go ahead, check 'em)

- [ ] Merge destination is correct
- [ ] Tests cases have been added where appropriate (Jest, Loki, Cypress)
- [ ] Code is correct as understood and conforms to quality standards
- [ ] Acceptance criteria have been met

## Testing Instructions

Before testing this build you will need to have postgres installed and have a database already created.
Please look at the readme to see how to set up your environment variables.

1. Checkout branch feature/bcrypt
2. If you have tables built in your database run `npx sequelize db:migrate:undo:all`, if not skip this step
3. Run `npx sequelize db:migrate`
4. Run `npx sequelize db:seed:all`
5. Log in at http://localhost:YOUR_PORT/auth/login
6. Email: admin@gmail.com,  password:password
7. You can create your own user at http://localhost:YOUR_PORT/auth/signup

### CODE REVIEW

File: src/auth/index.ts [hunk 1]
```diff
@@ -21,29 +24,77 @@ router.post("/login", async function(req, res) {
...
+      });
   } else {
     res.status(401).json({
-      message: "Authorization Failed ⛔️"
+      message: "Failed to login"
     });
   }
 });
+
+router.post("/signin", async function(req, res) {
```
Comment (blaketarter) [inline src/auth/index.ts]: @Alexandria Usually `signin` is the same kind of route that `login` is—they both start a new session for a user that already exists. Most APIs will just use one term or the other. Where `signup` is usually the route for creating a new user in the database and starting a session along with it.

File: src/auth/index.ts [hunk 2]
```diff
@@ -21,29 +24,77 @@ router.post("/login", async function(req, res) {
...
+  await validateLogin(req.body).catch(() => {
+    res.status(401).json({
+      message: "Authorization Faild: Invalid information ⛔️"
+    });
+  });
+
+  await User.findAll({
```
Comment (blaketarter) [inline src/auth/index.ts]: @Alexandria You might not need this `await` since there is no code being executed after this call.

**Review by blaketarter — COMMENTED** [review]
Looks good for the most part, a few comments though.

File: src/auth/index.ts (hunk 1 above)
Comment (Alexandria) [reply src/auth/index.ts]: Ha, this is definitely a case of me overthinking  the naming conventions..

**Review by blaketarter — APPROVED** [review]
👍

**Review by kileybublitz — APPROVED** [review]

**Pull request merged**